src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -245,10 +245,27 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
                     var assembly = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
                     if (assembly != null)
                     {
-                        AddRange(allSymbolReferences, await finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact).ConfigureAwait(false));
+                        findTasks.Add(finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact));
                     }
                 }
+            }
+
+            while (findTasks.Count > 0)
+            {
+                // Keep on looping through the 'find' tasks, processing each when they finish.
+                cancellationToken.ThrowIfCancellationRequested();
+                var doneTask = await Task.WhenAny(findTasks).ConfigureAwait(false);
```

Comment:
So none of findTasks actually have a cancellationToken? It might be worth moving your comment up a bit to clarify this, since at first this threw me off.